node/consensus/consensus.go

```diff
@@ -564,3 +571,24 @@ func (c *Consensus) verifyCE(req []byte) (smartbft_types.RequestInfo, *state.Con
 		return smartbft_types.RequestInfo{}, ce, fmt.Errorf("empty control event")
 	}
 }
+
+func (c *Consensus) validateRouterFromContext(ctx context.Context) error {
+	// extract the client certificate from the context
+	cert := node.ExtractCertificateFromContext(ctx)
+	if cert == nil {
+		return errors.New("error: access denied; could not extract certificate from context")
+	}
+
+	// extract the router certificate from the ConsenterNodeConfig
+	rawRouterCert := c.Config.Router.TLSCert
+	pemBlock, _ := pem.Decode(rawRouterCert)
+	if pemBlock == nil || pemBlock.Bytes == nil {
+		return fmt.Errorf("error decoding router TLS certificate")
+	}
+
+	// compare the two certificates
+	if !bytes.Equal(pemBlock.Bytes, cert.Raw) {
+		return fmt.Errorf("error: access denied. The client certificate does not match the router's certificate")
```

Comment:
Here, because it is a config or security problem, log the details of the offending incoming cert, in human readable format as well as raw. This may help correct the issue.